### PR TITLE
TomDoc methods in tutorial.rb #275

### PR DESCRIPTION
**Pull request by bbiehl**
- [x] privacy_check
- [x] draft_check
- [x] get_privacy_notice
- [x] get_category_notice
- [x] get_draft_title_for_tutorial
- [x] create_chapters(params)
- [x] save_if_valid

### CODE REVIEW

File: app/models/tutorial.rb [hunk 1]
```diff
@@ -12,6 +12,17 @@ class Tutorial < ActiveRecord::Base
...
+  # Checks a user's access to tutorial.
+  #
+  # If the tutorial is not public.
+  #   Sets user by finding tutorial's user_id
+  #   If user's company_id does not match logged in user's company_id.
+  #     user can not view tutorial.
+  #   Else the user's company_id matches logged in user's company_id.
+  #     user can view tutorial
+  # Else the tutorial is public and the user can view the tutorial.
+  #
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 2]
```diff
@@ -30,6 +41,7 @@ def privacy_check(user_check)
 
   end
 
+  #
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 3]
```diff
@@ -30,6 +41,18 @@ def privacy_check(user_check)
...
   end
 
+  # Gets the draft status of a tutorial.
+  #
+  # Sets the following:
+  #  1. Draft status of tutorial - Boolean via self table.
+  #  2. Draft chapters Array is empty? - Boolean via chapters table.
+  #  3. Draft steps Array is empty? - Boolean via steps table.
+  #
+  # If tutorial is a draft, or tutorial has draft chapters, or tutorial has draft steps.
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Line is too long. [88/80]

File: app/models/tutorial.rb [hunk 4]
```diff
@@ -30,6 +41,18 @@ def privacy_check(user_check)
...
+  # => returns Boolean, draft_check == true
+  #
+  # If tutorial is not a draft or tutorial, or tutorial has no draft chapters, or tutorial has no draft steps.
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Line is too long. [110/80]

File: app/models/tutorial.rb [hunk 5]
```diff
@@ -42,6 +65,12 @@ def draft_check
     end
   end
 
+  # Gets a privacy notice for a tutorial.
+  #
+  # Returns a String "Public" or "Private" based on Boolean value of tutorial's public setting.
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Line is too long. [95/80]

File: app/models/tutorial.rb [hunk 6]
```diff
@@ -42,6 +65,12 @@ def draft_check
...
+  #
+  # If tutorial is public, Returns "Public" String.
+  # If tutorial is not public, Returns "Private" String.
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 7]
```diff
@@ -30,6 +41,22 @@ def privacy_check(user_check)
...
   end
 
+  # Gets the draft status of a tutorial.
+  #
+  # Sets the following:
+  #  1. Draft status of tutorial - Boolean via self table.
+  #  2. Draft chapters Array is empty? - Boolean via chapters table.
+  #  3. Draft steps Array is empty? - Boolean via steps table.
+  #
+  # If tutorial is a draft,
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 8]
```diff
@@ -30,6 +41,22 @@ def privacy_check(user_check)
...
+  #  1. Draft status of tutorial - Boolean via self table.
+  #  2. Draft chapters Array is empty? - Boolean via chapters table.
+  #  3. Draft steps Array is empty? - Boolean via steps table.
+  #
+  # If tutorial is a draft, 
+  # or tutorial has draft chapters,
+  # or tutorial has draft steps.
+  # => returns Boolean, draft_check == true
+  #
+  # If tutorial is not a draft or tutorial,
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 9]
```diff
@@ -30,6 +41,22 @@ def privacy_check(user_check)
...
+  #  2. Draft chapters Array is empty? - Boolean via chapters table.
+  #  3. Draft steps Array is empty? - Boolean via steps table.
+  #
+  # If tutorial is a draft, 
+  # or tutorial has draft chapters,
+  # or tutorial has draft steps.
+  # => returns Boolean, draft_check == true
+  #
+  # If tutorial is not a draft or tutorial, 
+  # or tutorial has no draft chapters,
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 10]
```diff
@@ -42,10 +69,21 @@ def draft_check
...
+  # Returns a String "Public" or "Private".
+  #
+  # If tutorial is public, Returns "Public" String.
+  # If tutorial is not public, Returns "Private" String.
   def get_privacy_notice
     return self.public == true ? "Public" : "Private"
   end
 
+  # Gets a category notice.
+  #
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 11]
```diff
@@ -42,10 +69,21 @@ def draft_check
...
+  # If tutorial is public, Returns "Public" String.
+  # If tutorial is not public, Returns "Private" String.
   def get_privacy_notice
     return self.public == true ? "Public" : "Private"
   end
 
+  # Gets a category notice.
+  # 
+  # Initially Sets notice = empty String.
+  #
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 12]
```diff
@@ -42,10 +69,21 @@ def draft_check
...
+  # If tutorial is not public, Returns "Private" String.
   def get_privacy_notice
     return self.public == true ? "Public" : "Private"
   end
 
+  # Gets a category notice.
+  # 
+  # Initially Sets notice = empty String.
+  # 
+  # If tutorial's category
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 13]
```diff
@@ -63,6 +101,7 @@ def get_category_notice
     return notice
   end
 
+  #
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 14]
```diff
@@ -42,10 +69,27 @@ def draft_check
...
+  # Gets a tutorial's category notice.
+  #
+  # Initially Sets notice = empty String.
+  #
+  # If tutorial's category is equal to String "1",
+  # => Sets notice String equal to "CATEGORY 1".
+  # If tutorial's category is equal to String "2",
+  # => Sets notice String equal to "CATEGORY 2.""
+  # etc...
+  #
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 15]
```diff
@@ -62,7 +106,7 @@ def get_category_notice
 
     return notice
   end
-
+
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 16]
```diff
@@ -63,22 +99,33 @@ def get_category_notice
...
     return self.draft == true ? "DRAFT: #{self.title.upcase}" : "EDIT: #{self.title.upcase}"
   end
 
-def create_chapters(params)
+  # Creates new chapters.
+  #
+  # Takes params arguments.
+  # 
+  # Returns a new Chapter Object.
+  def create_chapters(params)
```
Comment (sumeetjain) [inline app/models/tutorial.rb:119]: Remove unused methods (tutorial#create_chapters)

File: app/models/tutorial.rb [hunk 17]
```diff
@@ -63,22 +99,33 @@ def get_category_notice
...
-      @new_chapter = self.chapters.new(title: params[:title], description: params[:description], draft: params[:draft])
+    @new_chapter = self.chapters.new(title: params[:title], description: params[:description], draft: params[:draft])
       #@new_chapters << chapter 
-    end
+  end
 
-def save_if_valid
-  if @new_chapter.valid?
-    @new_chapter.save
+  def save_if_valid
```
Comment (sumeetjain) [inline app/models/tutorial.rb:129]: Remove unused methods (tutorial#save_if_valid)

File: app/models/tutorial.rb [hunk 18]
```diff
@@ -63,22 +99,33 @@ def get_category_notice
...
+  # => Returns a String "EDIT: tutorial.title.upcase"
   def get_draft_title_for_tutorial
     return self.draft == true ? "DRAFT: #{self.title.upcase}" : "EDIT: #{self.title.upcase}"
   end
 
-def create_chapters(params)
+  # Creates new chapters.
+  #
+  # Takes params arguments.
+  #
```
Comment (sumeetjain) [inline app/models/tutorial.rb:117]: Trailing whitespace detected.

File: app/models/tutorial.rb [hunk 19]
```diff
@@ -63,22 +99,33 @@ def get_category_notice
...
+  # Creates new chapters.
+  #
+  # Takes params arguments.
+  # 
+  # Returns a new Chapter Object.
+  def create_chapters(params)
     #@new_chapters = []
     #params.each do |key, value|
-      @new_chapter = self.chapters.new(title: params[:title], description: params[:description], draft: params[:draft])
+    @new_chapter = self.chapters.new(title: params[:title], description: params[:description], draft: params[:draft])
```
Comment (sumeetjain) [inline app/models/tutorial.rb:122]: Line is too long. [117/80]

File: app/models/tutorial.rb [hunk 20]
```diff
@@ -63,22 +99,33 @@ def get_category_notice
...
       #@new_chapters << chapter 
-    end
+  end
 
-def save_if_valid
-  if @new_chapter.valid?
-    @new_chapter.save
+  def save_if_valid
+    if @new_chapter.valid?
```
Comment (sumeetjain) [inline app/models/tutorial.rb:130]: Use a guard clause instead of wrapping the code inside a conditional expression.

Comment (sumeetjain) [inline app/models/tutorial.rb:130]: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.

**Comment by bbiehl** [comment]
Based on the time left to work on this, I will have to merge.  This is not the best it could be but is at least something.  Might update TomDoc'ing whence the testing commences.

**Pull request merged**